lms/djangoapps/courseware/courses.py

```diff
@@ -431,45 +431,62 @@ def date_block_key_fn(block):
     return block.date or datetime.max.replace(tzinfo=pytz.UTC)
 
 
-def get_course_assignment_due_dates(course, user, request, num_return=None,
-                                    include_past_dates=False, include_access=False):
+def get_course_assignment_date_blocks(course, user, request, num_return=None,
+                                      include_past_dates=False, include_access=False):
```

Comment:
This and the rest of the changes in this file are just to separate the logic of getting-info-from-course and dumping-that-info-into-date-blocks. Since my ics generator also wants the same info-from-course stuff.